client/src/sentiment.js

```diff
@@ -0,0 +1,170 @@
+const fs = require('fs');
+const Sentiment = require('sentiment');
+
+const { dataStoreFilename } = require('./constant');
+const { removeHtmlTags } = require('./parse-utils');
+
+const storeData = require(dataStoreFilename);
+
+const range = (n) => Array(n).fill(undefined).map((_, i) => i);
+// What does this function do?
+//Looks like it returns an array with the index as the values
+//why fill with undefined?
+
+const commonWords = [
+	'---',
+	'a',
+	'about',
+	'after',
+	'and',
+	'are',
+	'as',
+	'asked',
+	'at',
+	'be',
+	'but',
+	'by',
+	'could',
+	'for',
+	'from',
+	'go',
+	'going',
+	'has',
+	'have',
+	'he',
+	'her',
+	'his',
+	'i',
+	'in',
+	'is',
+	'it',
+	'its',
+	'may',
+	'most',
+	'new',
+	'no',
+	'not',
+	'now',
+	'of',
+	'on',
+	'or',
+	'other',
+	'our',
+	'said',
+	'she',
+	'take',
+	'that',
+	'the',
+	'their',
+	'they',
+	'this',
+	'to',
+	'us',
+	'was',
+	'we',
+	'were',
+	'what',
+	'while',
+	'who',
+	'will',
+	'with',
+	'would',
+	'you',
+	'your'
+];
+
+const isNonPhrase = (phrase) => {
+	const words = phrase.split(' ');
+	if (words.every((word) => commonWords.includes(word))) {
+		return true;
+	}
+	return false;
+};
+
+const splitIntoWords = (text) => text.split(/\s|---?/);
+
+const cleanWord = (word) =>
+	word
+		.replace('—', '---')
+		.replace(/’/g, "'")
+		.replace(/'s$/, '')
+		.replace(/'$/g, '')
+		.replace(/[–;…!?.,:"“”]/g, '')
+		.toLowerCase();
+
+const cleanText = (text) => text.replace(/&amp;/g, '&').replace(/&nbsp;/g, ' ');
+
+const ngrams = (text, n) => {
+	const words = splitIntoWords(text).map(cleanWord);
+	return range(words.length - n + 1).map((i) => range(n).map((j) => words[i + j]));
+};
+
+const collectKeyPhrases = (text, count = 10) => {
+	return [
+		1,
+		2,
+		3
+	]
+		.reduce((ngramsRecord, n) => {
+			const record = ngrams(text, n).reduce((record, ngram) => {
+				const phrase = ngram.join(' ');
+				if (!phrase) return record;
+				if (!record[phrase]) record[phrase] = 1;
+				else record[phrase]++;
+				return record;
+			}, {});
+			return ngramsRecord.concat(
+				Object.entries(record).filter(
+					(
+						[
+							phrase,
+							_occurences
+						]
```

Comment:
I'm using Prettier in my setup. Can you add a husky setup with `{"precommit": ['prettier --write', 'git add']}`?

Reply:
?? I've got Prettier as well. Is it an issue with the configuration? I'm not sure what you mean by husky setup.

Reply:
I never added a configuration. I'm just using prettier defaults.

Husky is a javascript-y way of doing git hooks. If you use prettier, don't worry about it. It doesn't look like you're using prettier

Reply:
I just took a closer look now. Your prettier config uses tabs instead of spaces